docs/framework-design.md

```diff
@@ -0,0 +1,108 @@
+Version-independent build and test framework
+============================================
+
+## Introduction
+
+The [`mbedtls-framework`](https://github.com/Mbed-TLS/mbedtls-framework) repository provides tooling used to build and test TF-PSA-Crypto (all versions) and Mbed TLS from 3.6.0 onwards.
+
+## Requirements
+
+### Initial motivation
+
+Mbed TLS 3.x was a library for cryptography, X.509 and TLS. In 2024, the cryptography part of the library moved to a separate repository: [TF-PSA-Crypto](https://github.com/Mbed-TLS/TF-PSA-Crypto). Many support files are used by both projects: several helper scripts in `scripts` and `tests/scripts`, most of the test data in `tests/data`, a few helper programs under `programs`, most of the test helper code in `tests/include`, `tests/src` and `tests/drivers`, etc.
+
+The Mbed TLS project maintains long-time support (LTS) branches (with only bug fixes) in addition to the `development` branch where new features are added. Fixes to bugs often need to be backported from `development` to LTS branches, which involves backporting tests, which often involves backporting test helper code. This would be facilitated by hosting the test helper code in a shared repository that can be consumed by all maintained branches, reducing the amount of code to backport.
```

Comment:
Actually I have a problem with your wording. This is in the “Initial motivation” section, and this paragraph explains the old situation. The next paragraph introduces the framework repository. At this point in the explanation, there is no separate repository for helper code: we're explaining why we wanted to create one. I will try to simplify this bit though.

Reply:
Ah okay, that makes sense. I think my confusion is because it could be read as "this is what we currently do / have done in the past":

"This would [previously] be facilitated by..."

Perhaps a wording more like: "This situation could be improved by hosting..." would be cleaner?